src/web/mjs/connectors/TuMangaOnlineHentai.mjs

```diff
@@ -22,81 +22,38 @@ export default class TuMangaOnlineHentai extends Connector {
         return new Manga(this, id, title);
     }
 
-    /**
-     *
-     */
-    _getMangaListFromPages( mangaPageLinks, index ) {
-        index = index || 0;
-        let request = new Request( mangaPageLinks[ index ], this.requestOptions );
-        return this.fetchDOM( request, 'div.panel-body table.table tbody tr td.text-left a', 5 )
-            .then( data => {
-                let mangaList = data.map( element => {
-                    return {
-                        id: element.href.split( '/' ).pop(),
-                        title: element.text.trim()
-                    };
-                } );
-                if( index < mangaPageLinks.length - 1 ) {
-                    return this._getMangaListFromPages( mangaPageLinks, index + 1 )
-                        .then( mangas => mangaList.concat( mangas ) );
-                } else {
-                    return Promise.resolve( mangaList );
-                }
-            } );
+    async _getMangas() {
+        let mangaList = [];
+        let hasNext = true;
+        let page = 1;
+        while (hasNext) {
+            let request = new Request(this.url + '/section/all?view=list&order=alphabetic&page=' + page, this.requestOptions);
+            const body = (await this.fetchDOM(request, 'body'))[0];
+            let data = [...body.querySelectorAll('div.panel-body table.table tbody tr td.text-left a')];
+            let mangas = data.map(element => {
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(element, this.url).replace('/contents/', ''),
+                    title: element.text.trim()
+                };
+            });
+            mangaList = [...mangaList, ...mangas];
```

Comment:
May use `mangaList.push(...mangas)`

Reply:
I didn't know that push would merge the array. I though it would insert an single item of type array.